client/tests/integration/events/data.rs

```diff
@@ -138,7 +138,7 @@ fn transaction_execution_should_produce_events(
     let listener = client.clone();
     let (init_sender, init_receiver) = mpsc::channel();
     let (event_sender, event_receiver) = mpsc::channel();
-    let event_filter = DataEventFilter::AcceptAll.into();
+    let event_filter = FilterBox::Data(DataEventFilter::ByAny);
```

Comment:
```suggestion
    let event_filter = DataEventFilter::ByAny;
```

and change `listen_for_events` to take `Into<FilterBox>`

Reply:
if we can avoid using `FilterBox` in the public API we should. `***Box` structures exist for serialization and occasional grouping

Reply:
> if we can avoid using FilterBox in the public API we should

It would be nice, but I don't see how we could do that with event filters... Do you have an idea in mind?

Reply:
In this example we can. We should change the functions in the client that take `FilterBox` to take `Into<FilterBox>`

Reply:
I think I get the idea.

I'll do this for `listen_for_events`, but for `Action::new` I would like to make `Action` and `Filter` non-generic, so that we wouldn't have it accept `Into<F>`. Filed #4259 for it